src/coreclr/jit/jitconfig.cpp

```diff
@@ -361,6 +361,26 @@ bool JitConfigValues::MethodSet::contains(const char*       methodName,
                         return true;
                     }
                 }
+
+                // Special case: Main for top-level statements
+                if ((strcmp(expectedMethodName, "Main") == 0) && (strcmp(methodName, "<Main>$") == 0))
+                {
+                    return true;
+                }
```

Comment:
I personally find it quite odd to special case C# mangled function names like this in the JIT.

Reply:
There is a 45 line barebones regex implementation in eeconfig.cpp that I think we can just duplicate in the JIT.

Reply:
@stephentoub you mean for JitDisasm to always work as `Contains`? It might produce a lot of false positives e.g.
JitDisasm=Get -- will print out hundreds of methods etc

or you mean `JitDisasm=*Main*` (looks weird to me, especially for just Main which is single)

Reply:
> or you mean JitDisasm=*Main*

that

Reply:
Ok, implemented wildcards